editor/editor_storage.cpp

```diff
@@ -0,0 +1,72 @@
+#include "editor/editor_storage.hpp"
+
+#include "platform/platform.hpp"
+
+#include "coding/internal/file_data.hpp"
+
+#include "base/logging.hpp"
+
+#include "std/string.hpp"
+
+#include "3party/pugixml/src/pugixml.hpp"
+
+using namespace pugi;
+
+namespace
+{
+constexpr char const * kEditorXMLFileName = "edits.xml";
+string GetEditorFilePath() { return GetPlatform().WritablePathForFile(kEditorXMLFileName); }
+}  // namespace
+
+namespace editor
+{
+// StorageLocal ------------------------------------------------------------------------------------
+bool StorageLocal::Save(xml_document const & doc)
+{
+  auto const editorFilePath = GetEditorFilePath();
+  return my::WriteToTempAndRenameToFile(editorFilePath, [&doc](string const & fileName) {
+    return doc.save_file(fileName.data(), "  ");
+  });
+}
+
+bool StorageLocal::Load(xml_document & doc)
+{
+  auto const editorFilePath = GetEditorFilePath();
+  auto const result = doc.load_file(editorFilePath.c_str());
+  // Note: status_file_not_found is ok if a user has never made any edits.
+  if (result != status_ok && result != status_file_not_found)
+  {
+    LOG(LERROR, ("Can't load map edits from disk:", editorFilePath));
+    return false;
+  }
+
+  return true;
+}
+
+void StorageLocal::Reset()
+{
+  my::DeleteFileX(GetEditorFilePath());
+}
+
+// StorageMemory -----------------------------------------------------------------------------------
+StorageMemory::StorageMemory()
+  : m_doc(make_unique<xml_document>())
+{}
+
+bool StorageMemory::Save(xml_document const & doc)
+{
+  m_doc->reset(doc);
```

Comment:
By the way, is here a real need in unique_ptr? Why not to store m_doc as is, without unique_ptr wrapper?